### PR TITLE
Python 3: fix BGP

### PR DESCRIPTION
**Pull request by gpotter2**

### CODE REVIEW

File: scapy/contrib/bgp.py [hunk 1]
```diff
@@ -136,14 +136,14 @@ def addfield(self, pkt, s, val):
...
         return s[length:], self.m2i(pkt, s[:length])
 
     def m2i(self, pkt, m):
-        mask = struct.unpack(">B", m[0])[0]
+        mask = orb(m[0])
         mask2iplen_res = self.mask2iplen(mask)
-        ip = "".join(
-            [m[i + 1] if i < mask2iplen_res else b"\x00" for i in range(4)])
+        ip = b"".join(
+            [chb(m[i + 1]) if i < mask2iplen_res else b"\x00" for i in range(4)])
```
Comment (p-l-) [inline scapy/contrib/bgp.py]: This is a useless list. Plus we can skip the `chb()` call:
```Python
    ip = b"".join(m[i + 1:i + 2] if i < amsk2iplen_res else b"\x00" for i in range(4))
```

File: scapy/contrib/bgp.py [hunk 2]
```diff
@@ -181,13 +181,13 @@ def addfield(self, pkt, s, val):
...
+        length = self.mask2iplen(orb(s[0])) + 1
         return s[length:], self.m2i(pkt, s[:length])
 
     def m2i(self, pkt, m):
-        mask = struct.unpack(">B", m[0])[0]
-        ip = "".join(
-            [m[i + 1] if i < self.mask2iplen(mask) else b"\x00" for i in range(16)])
+        mask = orb(m[0])
+        ip = b"".join(
+            [chb(m[i + 1]) if i < self.mask2iplen(mask) else b"\x00" for i in range(16)])
```
Comment (p-l-) [inline scapy/contrib/bgp.py]: Same here.

Comment (gpotter2) [reply scapy/contrib/bgp.py]: Updated

File: scapy/contrib/bgp.py [hunk 3]
```diff
@@ -181,13 +180,12 @@ def addfield(self, pkt, s, val):
...
-        length = self.mask2iplen(struct.unpack(">B", s[0])[0]) + 1
+        length = self.mask2iplen(orb(s[0])) + 1
         return s[length:], self.m2i(pkt, s[:length])
 
     def m2i(self, pkt, m):
-        mask = struct.unpack(">B", m[0])[0]
-        ip = "".join(
-            [m[i + 1] if i < self.mask2iplen(mask) else b"\x00" for i in range(16)])
+        mask = orb(m[0])
+        ip = b"".join(m[i + 1] if i < self.mask2iplen(mask) else b"\x00" for i in range(16))
```
Comment (p-l-) [inline scapy/contrib/bgp.py]: Will this work with Python 3? I think you need `m[i + 1:i + 2]`, or did I miss something?

File: scapy/contrib/bgp.py [hunk 4]
```diff
@@ -136,14 +136,13 @@ def addfield(self, pkt, s, val):
...
+        length = self.mask2iplen(orb(s[0])) + 1
         return s[length:], self.m2i(pkt, s[:length])
 
     def m2i(self, pkt, m):
-        mask = struct.unpack(">B", m[0])[0]
+        mask = orb(m[0])
         mask2iplen_res = self.mask2iplen(mask)
-        ip = "".join(
-            [m[i + 1] if i < mask2iplen_res else b"\x00" for i in range(4)])
+        ip = b"".join(m[i + 1] if i < mask2iplen_res else b"\x00" for i in range(4))
```
Comment (p-l-) [inline scapy/contrib/bgp.py]: Will this work with Python 3? I think you need `m[i + 1:i + 2]`, or did I miss something?

File: scapy/contrib/bgp.py (hunk 3 above)
Comment (p-l-) [inline scapy/contrib/bgp.py]: Same here.

File: scapy/contrib/bgp.py (hunk 4 above)
Comment (gpotter2) [reply scapy/contrib/bgp.py]: my bad

**Pull request merged**